docker/base-runner.sh

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env bash
+
+host_architecture() {
+    # there's numerous compatibility modes, so we want
+    # to ensure that these are valid. we also want to
+    # use dpkg if it's available since it gives hard-float
+    # information on compatible architectures
+    local host
+    local arch
+    if dpkg >/dev/null 2>&1; then
+        host=$(dpkg --print-architecture)
+        arch="${host}"
+    else
```

Comment:
Normalize to Debian, because it holds slightly more information than `uname -m`, and all the supported architectures that Docker can run images on natively are Debian-supported targets.